File: app/src/main/java/cat/xojan/random1/domain/repository/ProgramRepository.java
package cat.xojan.random1.domain.repository;

import java.io.IOException;
import java.util.List;

import cat.xojan.random1.domain.entities.Podcast;
import cat.xojan.random1.domain.entities.Program;
import rx.Observable;

public interface ProgramRepository {

    List<Program> getProgramList() throws IOException;

    Observable<List<Podcast>> getPodcastByProgram(String programId) throws IOException;

    Observable<List<Podcast>> getPodcastBySection(String programId, String sectionId)
            throws IOException;
}

File: app/src/main/java/cat/xojan/random1/domain/repository/ProgramRepository.kt
package cat.xojan.random1.domain.repository

import cat.xojan.random1.domain.model.Program
import cat.xojan.random1.domain.model.Section
import io.reactivex.Single

interface ProgramRepository {
    fun getPrograms(refresh: Boolean): Single<List<Program>>

    fun getProgram(programId: String): Single<Program>

    fun hasSections(programId: String): Single<Boolean>

    fun getSections(programId: String): Single<List<Section>>
}